common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/MemoryRetriever.scala

```diff
@@ -1,8 +1,10 @@
 package uk.ac.wellcome.pipeline_storage
 
+import scala.collection.mutable
 import scala.concurrent.{ExecutionContext, Future}
 
-class MemoryRetriever[T](index: Map[String, T] = Map.empty)(
+class MemoryRetriever[T](val index: mutable.Map[String, T] =
+                           mutable.Map[String, T]())(
```

Comment:
nit: I think it makes more sense for this data to be immutable, we always know what is going inside it on creation in the tests. Not a big issue of course, just would confuse be a bit (a bit unexpected, unlike `MemoryIndexer` where the data actually changes through it's behaviour)

Reply:
Yeah, I’d definitely like to come back and fix this at some point. It was easier to keep this PR manageable – previously we'd write stuff into a VHS as we were going along.